tests/scripts/generate_test_code.py

```diff
@@ -846,6 +846,14 @@ def write_dependencies(out_data_f, test_dependencies, unique_dependencies):
     return dep_check_code
 
 
+INT_VAL_REGEX = re.compile(r'-?(\d+|0x[0-9a-f]+)$', re.I)
+def val_is_int(val: str) -> bool:
+    """Whether val is suitable as an 'int' parameter in the .datax file."""
+    if not INT_VAL_REGEX.match(val):
+        return False
+    # Limit the range to what is guaranteed to get through strtol()
+    return abs(int(val, 0)) <= 0x7fffffff
```

Comment:
As the comment indicates, currently, the value needs to be supported by `strtol`. 0x7fffffff is the largest value that is guaranteed to fit in `long`. `long` can be a 32-bit type (and it usually _is_ a 32-bit type on 32-bit architectures, and also on 64-bit Windows).

Reply:
Well, we don't have to use `strtol()` - we could use `strtoll()` or even write a simple conversion routine for ourselves (this is in the test code after all) - we only need bases 10 and 16

Reply:
Actually, because it's test code, we needn't bother reimplement the standard library.

And anyway that doesn't matter. The exp path has to exist anyway (so we can use any constant expression, including macro names that the python code could not recognize). So we might as well use it. The int path is just an optimization to reduce the size of the generated code.

Reply:
> As the comment indicates, currently, the value needs to be supported by `strtol`. 0x7fffffff is the largest value that is guaranteed to fit in `long`. `long` can be a 32-bit type (and it usually _is_ a 32-bit type on 32-bit architectures, and also on 64-bit Windows).

This makes sense to me. I was assuming `LONG_MAX(0x7fffffffffffffff)` is the largest value to fit in `long` but I didn't consider it could be a 32-bit type.